### PR TITLE
make cppcheck even more happier

### PR DESCRIPTION
**Pull request by chipitsine**
[channels/tsmf/client/gstreamer/tsmf_X11.c:317] -> [channels/tsmf/client/gstreamer/tsmf_X11.c:322]: (warning) Either the condition '!decoder' is redundant or there is possible null pointer dereference: decoder.
[channels/tsmf/client/gstreamer/tsmf_X11.c:470] -> [channels/tsmf/client/gstreamer/tsmf_X11.c:475]: (warning) Either the condition '!decoder' is redundant or there is possible null pointer dereference: decoder.
[channels/tsmf/client/gstreamer/tsmf_X11.c:472] -> [channels/tsmf/client/gstreamer/tsmf_X11.c:475]: (warning) Either the condition '!decoder' is redundant or there is possible null pointer dereference: decoder.
[channels/tsmf/client/tsmf_media.c:179] -> [channels/tsmf/client/tsmf_media.c:181]: (warning) Either the condition '!stream' is redundant or there is possible null pointer dereference: stream.
[client/Windows/wf_cliprdr.c:2219] -> [client/Windows/wf_cliprdr.c:2222]: (warning) Either the condition '!formatDataResponse' is redundant or there is possible null pointer dereference: formatDataResponse
[client/Windows/wf_cliprdr.c:2445] -> [client/Windows/wf_cliprdr.c:2448]: (warning) Either the condition '!fileContentsResponse' is redundant or there is possible null pointer dereference: fileContentsResponse.
[client/X11/xf_cliprdr.c:911] -> [client/X11/xf_cliprdr.c:913]: (warning) Either the condition '!clipboard' is redundant or there is possible null pointer dereference: clipboard.
[client/X11/xf_graphics.c:504] -> [client/X11/xf_graphics.c:506]: (warning) Either the condition '!xfc' is redundant or there is possible null pointer dereference: xfc.
[libfreerdp/core/transport.c:861] -> [libfreerdp/core/transport.c:863]: (warning) Either the condition '!transport' is redundant or there is possible null pointer dereference: transport.
[server/shadow/shadow_server.c:777] -> [server/shadow/shadow_server.c:791]: (warning) Either the condition '!server' is redundant or there is possible null pointer dereference: server.
[server/shadow/shadow_server.c:778] -> [server/shadow/shadow_server.c:791]: (warning) Either the condition '!server' is redundant or there is possible null pointer dereference: server.
[server/shadow/shadow_server.c:779] -> [server/shadow/shadow_server.c:791]: (warning) Either the condition '!server' is redundant or there is possible null pointer dereference: server.
[server/shadow/shadow_server.c:781] -> [server/shadow/shadow_server.c:791]: (warning) Either the condition '!server' is redundant or there is possible null pointer dereference: server.
[server/shadow/shadow_server.c:782] -> [server/shadow/shadow_server.c:791]: (warning) Either the condition '!server' is redundant or there is possible null pointer dereference: server.
[server/shadow/shadow_server.c:783] -> [server/shadow/shadow_server.c:791]: (warning) Either the condition '!server' is redundant or there is possible null pointer dereference: server.
[server/shadow/shadow_server.c:784] -> [server/shadow/shadow_server.c:791]: (warning) Either the condition '!server' is redundant or there is possible null pointer dereference: server.
[server/shadow/shadow_server.c:785] -> [server/shadow/shadow_server.c:791]: (warning) Either the condition '!server' is redundant or there is possible null pointer dereference: server.
[server/shadow/shadow_server.c:787] -> [server/shadow/shadow_server.c:791]: (warning) Either the condition '!server' is redundant or there is possible null pointer dereference: server.
[server/shadow/shadow_server.c:789] -> [server/shadow/shadow_server.c:791]: (warning) Either the condition '!server' is redundant or there is possible null pointer dereference: server.

### CODE REVIEW

File: client/X11/xf_cliprdr.c [hunk 1]
```diff
@@ -908,11 +908,12 @@ static BOOL xf_cliprdr_process_property_notify(xfClipboard* clipboard,
         XEvent* xevent)
 {
 	xfCliprdrFormat* format;
-	xfContext* xfc = clipboard->xfc;
 
 	if (!clipboard)
 		return TRUE;
 
+	xfContext* xfc = clipboard->xfc;
+
```
Comment (akallabeth) [inline client/X11/xf_cliprdr.c]: That will break older C compilers which require variables to be declared on top of a scope.

File: client/X11/xf_graphics.c [hunk 2]
```diff
@@ -501,11 +501,12 @@ BOOL xf_register_graphics(rdpGraphics* graphics)
 UINT32 xf_get_local_color_format(xfContext* xfc, BOOL aligned)
 {
 	UINT32 DstFormat;
-	BOOL invert = xfc->invert;
 
 	if (!xfc)
 		return 0;
 
+	BOOL invert = xfc->invert;
+
```
Comment (akallabeth) [inline client/X11/xf_graphics.c]: Same here, split declaration and assignment please

File: libfreerdp/core/transport.c [hunk 3]
```diff
@@ -858,11 +858,12 @@ int transport_check_fds(rdpTransport* transport)
 	int recv_status;
 	wStream* received;
 	DWORD now = GetTickCount();
-	DWORD dueDate = now + transport->settings->MaxTimeInCheckLoop;
 
 	if (!transport)
 		return -1;
 
+	DWORD dueDate = now + transport->settings->MaxTimeInCheckLoop;
+
```
Comment (akallabeth) [inline libfreerdp/core/transport.c]: And again

File: client/X11/xf_cliprdr.c (hunk 1 above)
Comment (chipitsine) [inline client/X11/xf_cliprdr.c]: sorry for that, I supposed c99.
ok, I'll fix that

File: channels/tsmf/client/tsmf_media.c [hunk 4]
```diff
@@ -176,11 +176,12 @@ static TSMF_SAMPLE* tsmf_stream_pop_sample(TSMF_STREAM* stream, int sync)
 	TSMF_STREAM* s;
 	TSMF_SAMPLE* sample;
 	BOOL pending = FALSE;
-	TSMF_PRESENTATION* presentation = stream->presentation;
 
 	if (!stream)
 		return NULL;
 
+	TSMF_PRESENTATION* presentation = stream->presentation;
```
Comment (mfleisz) [inline channels/tsmf/client/tsmf_media.c]: That will break older C compilers which require variables to be declared on top of a scope.

File: client/X11/xf_cliprdr.c (hunk 1 above)
Comment (chipitsine) [inline client/X11/xf_cliprdr.c]: I seen "BOOL", which lead me to c99

**Comment by akallabeth** [comment]
@mfleisz Could you merge after @freerdp-bot finishes the test?

**Comment by chipitsine** [comment]
it is not dead initialization. cppcheck otherwise likes to complain about "uninitialized variables".
I'd like it to complain on real issues and not to complain on such things

**Comment by akallabeth** [comment]
@chihchun Well, that is a disagreement between cppcheck and clang then. But you're right, not a real issue.
@freerdp-bot test!

**Comment by akallabeth** [comment]
@bmiklautz looks like @freerdp-bot stopped testing and is dead in the water :/

**Comment by chipitsine** [comment]
I did force-push. Maybe I should close this PR and open new one instead ?

**Comment by akallabeth** [comment]
@chipitsine Not your fault, there seems to be a glitch between github and our ci

File: libfreerdp/core/transport.c [hunk 5]
```diff
@@ -857,12 +857,14 @@ int transport_check_fds(rdpTransport* transport)
 	int status;
 	int recv_status;
 	wStream* received;
-	DWORD now = GetTickCount();
+	DWORD now = 0;
```
Comment (mfleisz) [inline libfreerdp/core/transport.c]: Why did you remove GetTickCount here? This breaks the now < dueDate check in the loop condition ...

Comment (chipitsine) [inline libfreerdp/core/transport.c]: oops

File: libfreerdp/core/transport.c [hunk 6]
```diff
@@ -857,12 +857,14 @@ int transport_check_fds(rdpTransport* transport)
...
 	DWORD dueDate = now + transport->settings->MaxTimeInCheckLoop;
```
Comment (mfleisz) [inline libfreerdp/core/transport.c]: Unchecked access (tranpsort->....)

**Comment by chipitsine** [comment]
@freerdp-bot , c'mon ....

**Comment by bmiklautz** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test again...

**Comment by akallabeth** [comment]
@freerdp-bot test, come on...

**Comment by akallabeth** [comment]
@freerdp-bot test and do not stop!
